src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1023,10 +1023,10 @@ else if (fieldClass == URI.class) {
             result = "URI";
         }
         else if (fieldClass == Pattern.class) {
-            result = "Regular Expression";
+            result = "Pattern";
         }
         else if (fieldClass == Pattern[].class) {
-            result = "Regular Expressions";
+            result = "Patterns";
```

Comment:
I don't see this defined in issue. Following other patters, it should be `Pattern[]` and not plural.

@romani ping

Reply:
@gaurabdg , 
```
 ~/java/github/romani/checkstyle [master|✔] 
06:52 $ ag -Q "Pattern[]"
src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
1028:        else if (fieldClass == Pattern[].class) {
1248:            else if (fieldClass == Pattern[].class) {
1251:                    final Pattern[] newArray = new Pattern[collection.size()];
1365:                result = Pattern[].class;

src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
326:        for (Pattern pattern : new Pattern[]

src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java
397:    private static final Pattern[] EMPTY_PATTERN_ARRAY = new Pattern[0];
406:    private Pattern[] groups = EMPTY_PATTERN_ARRAY;
416:    private Pattern[] staticGroups = EMPTY_PATTERN_ARRAY;
871:        final Pattern[] patterns;
897:    private static int getGroupNumber(Pattern[] patterns, String name) {
954:    private static Pattern[] compilePatterns(String... packageGroups) {
955:        final Pattern[] patterns = new Pattern[packageGroups.length];
```
I do not see where we have array of Patten types in Checks public setters, please point to the problem.

Reply:
A lot of the properties setter have `String... ` as setter paramter and then is converted to `Pattern[]`/`List<Pattern>`. The  properties do not have consistent initiation, as do `Pattern` property. Some have `Pattern` as the parameter for setter, but some have `String` and later conveted to `Pattern`, although their property type in xdoc is `regexp`.

Reply:
We should care only about setters, it is how value come to Class. 
Type should be `String[]` in description there should all deails on how it is parsed to inner implementation.